modules/geolocation/geolocationWorker.py

```diff
@@ -0,0 +1,89 @@
+from modules.geolocation.geolocation import Geolocation
+import multiprocessing as mp
+# For exception handling
+import queue
+
+
+def exit_requested(requestQueue):
+    return not requestQueue.empty()
+
+
+def geolocation_worker(pause, exitRequest, pipelineIn, pipelineOut, pipelineOutLock):
+
+    print("Start Geolocation")
+    locator = Geolocation()
+
+    while True:
+
+        pause.acquire()
+        pause.release()
+
+        # Pixel coordinates of tents and plane data
+        coordinates = pipelineIn.get()
+        # TODO verify this
+        if (not coordinates):
+            continue
+
+        ret, location = locator.run_locator(coordinates)
+
+        # Something has gone wrong, skip
+        if (not ret):
+            continue
+
+        # External lock required as geolocationOutput will dump the queue
+        pipelineOutLock.acquire()
+        pipelineOut.put(location)
+        pipelineOutLock.release()
+
```

Comment:
Can you clarify what the use case is for a pipeline lock? Do we want to prevent the downstream process (assuming it would be geolocation_output) from getting the location before we put it in the queue? Or do we want it to wait until multiple locations are in the pipeline?

Reply:
In a normal queue, processes take turns putting in/taking out items, while in this case, "taking out items" is grabbing the entire queue and shaking everything out.

Reply:
Got it